src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,233 @@
+            newGTs.add(gb.make());
+        }
+        return newGTs;
+    }
+
+
+    //TODO: is this doing the same thing as the method immediately above????  Quite likely, yes!!!
+    /**
+     * Create the new GenotypesContext with the subsetted PLs and ADs
+     *
+     * @param originalGs               the original GenotypesContext
+     * @param originalAlleles          the original alleles
+     * @param allelesToUse             the subset of alleles to use with the new Genotypes
+     * @param assignGenotypes          assignment strategy for the (subsetted) PLs
+     * @return                         a new non-null GenotypesContext
+     */
+    public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
+                                                 final List<Allele> originalAlleles,
+                                                 final List<Allele> allelesToUse,
+                                                 final GenotypeAssignmentMethod assignGenotypes) {
+        final GenotypesContext newGTs = GenotypesContext.create(originalGs.size());
+        final Permutation<Allele> allelePermutation = new IndexedAlleleList<>(originalAlleles).permutation(new IndexedAlleleList<>(allelesToUse));
+
+        final Map<Integer, int[]> subsettedLikelihoodIndicesByPloidy = new TreeMap<>();
+        for (final Genotype g : originalGs.iterateInSampleNameOrder()) {
+            final int ploidy = g.getPloidy();
+            if (!subsettedLikelihoodIndicesByPloidy.containsKey(ploidy)) {
+                subsettedLikelihoodIndicesByPloidy.put(ploidy, subsettedGenotypeLikelihoodIndices(ploidy, originalAlleles, allelesToUse));
+            }
+            final int[] subsettedLikelihoodIndices = subsettedLikelihoodIndicesByPloidy.get(ploidy);
+
+            final int expectedNumLikelihoods = GenotypeLikelihoods.numLikelihoods(originalAlleles.size(), ploidy);
+            // create the new likelihoods array from the alleles we are allowed to use
+            double[] newLikelihoods = null;
+            if (g.hasLikelihoods()) {
+                final double[] originalLikelihoods = g.getLikelihoods().getAsVector();
+                newLikelihoods = originalLikelihoods.length == expectedNumLikelihoods ?
+                        MathUtils.normalizeFromLog10(Arrays.stream(subsettedLikelihoodIndices)
+                                .mapToDouble(idx -> originalLikelihoods[idx]).toArray(), false, true) : null;
+            }
+
+            final boolean useNewLikelihoods = newLikelihoods != null && GATKVariantContextUtils.isInformative(newLikelihoods);
+            final GenotypeBuilder gb = useNewLikelihoods ? new GenotypeBuilder(g).PL(newLikelihoods) : new GenotypeBuilder(g).noPL();
+
+            // make new genotype call
+            GATKVariantContextUtils.refreshGenotypeCall(g.getAlleles(), gb, assignGenotypes, newLikelihoods, allelesToUse);
```

Comment:
Make genotype call doesn't require an existing call and therefore doesn't work with the `BEST_MATCH` assignment method.  Otherwise they are the same.  I see in your comment below that you think `BEST_MATCH` is dubious and I feel the same way.  If I delete this one, which is never used in `HaplotypeCaller` as far as I can tell, I will be able to merge `refreshGenotypeCall` and `makeGenotypeCall`.